src/Columns/ColumnString.cpp

```diff
@@ -481,6 +482,26 @@ void ColumnString::updatePermutationWithCollation(const Collator & collator, Per
             DefaultPartialSort());
 }
 
+size_t ColumnString::estimateNumberOfDifferent(const Permutation & perm, const EqualRange & range, size_t /*samples*/) const
+{
+    // TODO: sample random elements
+    size_t range_size = getRangeSize(range);
+    if (range_size <= 1ULL)
```

Comment:
From my point of view, this is not a micro-optimization, since in most cases, in arbitrary tables, equivalence classes will be of size 1, and I do not want to do a lot of meaningless work for them.